bindings/go/constructor/construct.go

```diff
@@ -68,7 +68,7 @@ type ConstructorOrExternalComponent struct {
 	ExternalComponent    *descriptor.Descriptor
 }
 
-func (c *DefaultConstructor) Construct(ctx context.Context, componentConstructor *constructor.ComponentConstructor) ([]*descriptor.Descriptor, error) {
+func (c *DefaultConstructor) Construct(ctx context.Context, componentConstructor *constructor.ComponentConstructor) ([]*descriptor.Descriptor, *syncdag.SyncedDirectedAcyclicGraph[string], error) {
```

Comment:
why do we return a descriptor list on top of the graph? wont we need only the graph and can get the roots from there?

Reply:
for the sake of the scope of this task, I will keep the descriptors return for now